### ISSUE TITLE
documentation contribution guidelines

### ISSUE THREAD

**Issue by nowthey** [wibyweb/wiby#8]
Hi, I apologize for how long this is. Basically I am asking if you want documentation contributions. Also I am trying to give you a picture of where I am coming from because I most want to know if you want contribs *from me*. 

I found this project a couple of months ago and have been pretty jazzed about it since that time. I really agree with all the philosophy and goals of why you made it. I have been in search of exactly this for a long time. As far as I know, no one has filled this niche yet. 

That said I am only involved in this sort of thing on the weekends. Like it's not my job, I don't have training etc. So I have been slowly working through the instructions getting everything set up. Not my first adventure in selfhosting but it's a challenge. 

Finally I am at "Configure nginx for Wiby"--- so not quite done but pretty confident I will be able to work out what is required after a little break. 

I have been taking notes about what pieces were missing for me from the documentation as I went and I am wondering if you are interested in contributions that would clarify for a less experienced admin? I do not see any contrib guidelines. It also seems like a very low-activity repo, there are only 5 issues. Which I find really puzzling given how unique and useful wiby is. Has the project been cruising under the radar, or do you prefer to work on it more independently?

I would also say that, because I never make things simple for myself, I deviated from the instructions somewhat.  I have been working on my local manjaro system (really not a great server environment), and I decided to use MariaDB instead of MySQL after reading about them both and seeing MariaDB is more FLOSS.

So some of the problems I had were consequent of that, for example configuration files being located differently. I have in my notes how to locate the correct file on the system, which I think would be useful to include. Perhaps I could collect the notes regarding different environments in a separate page to avoid confusing the existing narrative. 

Other confusions would have come up in any case because it is obviously not a complete guide to every single thing about running a LEMP server. For example I took a *very* long time figuring out how to go about following "Login to MySQL and type:". It turns out the answer is in the *next* step but as I was going in order I didn't know that. Figuring the credentials was pretty elaborate; perhaps this could have been avoided with a slightly different initial set up. I can write how I got passed it (resetting to a known password) but I doubt my method is ideal.

I am also not sure how, when this is eventually up and running, updates would be handled. Today I realized that I have been working on it so slowly that you have made changes since I first cloned the repo. I hit a bug that prevented proceeding, but you actually already fixed it. :) So I pulled changes and repeated prior steps, like recompiling everything, but I wasn't sure if I needed to do that, or if it might break anything etc. Once I *have* a database with stuff in it, I guess I wouldn't want to repeat the step `mysql -u root -p wiby < wiby.sql`? But could it happen that changes to the application would be made requiring the db to be modified? 

**TLDR** So all in all, I am wondering, what is your plan/roadmap/vision for this project in particular the setup/documentation? And would you like input? 

If you are interested, I might try setting up a fresh Ubuntu VPS to weed out what issues are due to idiosyncrasies of using my personal computer with all its messiness. So I would start from scratch and follow everything as written, filling in gaps. 

BTW I understand that you probably don't want to get into a situation of hand-holding a lot of people who lack foundational skills to run something like this. Sometimes barriers to entry are OK and it is fair enough to manage expectations. 

thanks for sharing your work

**Comment by wibyweb** [comment]
Hi there, if you want to create an "install guide for dummies" in your github repo, I would be happy to place a link to this inside the official guide for additional help, as long as it is tailored to Ubuntu. 

As for the database issues you ran into, I don't anticipate needing to make further changes that could break your existing install *knock on wood*.

Regarding guidelines and contributions, I am for my own sanity running this project as an absolute dictatorship. So I don't have any guidelines at the moment. I just want to keep the software lightweight and easy to modify so others can create their own mods or supplemental docs.
